packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -63,39 +63,39 @@ export abstract class QueryBuilderExplorerTreeNodeData implements TreeNodeData {
   label: string;
   dndText: string;
   childrenIds: string[] = [];
-  mapped: boolean;
-  /**
-   * This flag is used to skip mappedness checking for the whole branch (i.e. every children of
-   * a node with this property set to `true` will also be skipped in mappedness checking).
-   * This is to facillitate complicated mappedness checkings that Studio would not attempt to handle:
-   * e.g. derived properties, operation class mappings, etc.
-   */
-  skipMappingCheck: boolean;
   isPartOfDerivedPropertyBranch: boolean;
   type: Type;
-  setImpl?: SetImplementation | undefined;
+  mappingData: PropertyMappingData;
 
   constructor(
     id: string,
     label: string,
     dndText: string,
-    mapped: boolean,
-    skipMappingCheck: boolean,
     isPartOfDerivedPropertyBranch: boolean,
     type: Type,
-    setImpl: SetImplementation | undefined,
+    mappingData: PropertyMappingData,
   ) {
     this.id = id;
     this.label = label;
     this.dndText = dndText;
-    this.mapped = mapped;
-    this.skipMappingCheck = skipMappingCheck;
     this.isPartOfDerivedPropertyBranch = isPartOfDerivedPropertyBranch;
     this.type = type;
-    this.setImpl = setImpl;
+    this.mappingData = mappingData;
   }
 }
 
+export type PropertyMappingData = {
+  mapped: boolean;
+  /**
+   * This flag is used to skip mappedness checking for the whole branch (i.e. every children of
+   * a node with this property set to `true` will also be skipped in mappedness checking).
+   * This is to facillitate complicated mappedness checkings that Studio would not attempt to handle:
+   * e.g. derived properties, operation class mappings, etc.
+   */
+  skipMappingCheck: boolean;
+  setImpl?: SetImplementation | undefined;
```

Comment:
Should we call this `targetSetImplementation`? what do you think?

Reply:
Yeah makes sense